pootle/apps/pootle_fs/files.py

```diff
@@ -0,0 +1,229 @@
+        self.store_fs.save()
+
+    def on_sync(self, latest_hash, revision):
+        """
+        Called after FS and Pootle have been synced
+        """
+        self.store_fs.resolve_conflict = None
+        self.store_fs.staged_for_merge = False
+        self.store_fs.last_sync_hash = latest_hash
+        self.store_fs.last_sync_revision = revision
+        self.store_fs.save()
+        logger.debug("File synced: %s" % self.path)
+
+    def pull(self, user=None):
+        """
+        Pull FS file into Pootle
+        """
+        current_hash = self.latest_hash
+        last_hash = self.store_fs.last_sync_hash
+        if self.store and last_hash == current_hash:
+            return
+        logger.debug("Pulling file: %s" % self.path)
+        if not self.store:
+            self.create_store()
+        self.sync_to_pootle(user=user)
+
+    def push(self, user=None):
+        """
+        Push Pootle ``Store`` into FS
+        """
+        if not self.store:
+            return
+        current_revision = self.store.get_max_unit_revision()
+        last_revision = self.store_fs.last_sync_revision
+        if self.exists and last_revision == current_revision:
+            return
+        logger.debug("Pushing file: %s" % self.path)
+        directory = os.path.dirname(self.file_path)
+        if not os.path.exists(directory):
+            logger.debug("Creating directory: %s" % directory)
+            os.makedirs(directory)
+        self.sync_from_pootle()
+
+    def read(self):
+        if not self.exists:
+            return
+        with open(self.file_path) as f:
+            return f.read()
+
+    def remove_file(self):
+        if self.exists:
+            os.unlink(self.file_path)
+
+    def deserialize(self):
+        if not self.exists:
+            return
+        deserialized = ""
+        with open(self.file_path) as f:
+            deserialized = f.read()
+        if self.store:
+            return self.store.deserialize(deserialized)
+        serialio = io.BytesIO(deserialized)
+        return getclass(serialio)(serialio.read())
+
+    def serialize(self):
+        if not self.store:
+            return
+        return self.store.serialize()
+
+    def sync_from_pootle(self):
+        """
+        Update FS file with the serialized content from Pootle ```Store```
+        """
+        with open(self.file_path, "w") as f:
+            f.write(self.serialize())
+        logger.debug("Pushed file: %s" % self.path)
+
+    def sync_to_pootle(self, pootle_wins=False, merge=False, user=None):
+        """
+        Update Pootle ``Store`` with the parsed FS file.
+        """
+        User = get_user_model()
+        resolve_conflict = (
+            pootle_wins
+            and store_models.POOTLE_WINS
+            or store_models.FILE_WINS)
+        if merge:
+            revision = self.store_fs.last_sync_revision
+        else:
+            revision = self.store.get_max_unit_revision() + 1
```

Comment:
that would increase the revision of the system though, whereas we just want to set the revision to something higher than the Store's.

That said, what would be cheaper than get_max_unit_revision is doing Revision.get()